src/algebra/big_operators/fin.lean

```diff
@@ -141,8 +141,37 @@ end fin
 
 namespace list
 
+section monoid
+
+variables [monoid α] {n : ℕ}
+
+/-- Given `f = (a₁, ..., aₙ)` in `αⁿ`, `partial_prod f` is `(1, a₁, a₁a₂, ..., a₁...aₙ)` in `αⁿ⁺¹`.
+-/
+@[to_additive "Given `f = (a₁, ..., aₙ)` in `αⁿ`, `partial_sum f` is
+`(1, a₁, a₁ + a₂, ..., a₁ + ... + aₙ)` in `αⁿ⁺¹`."]
+def partial_prod (f : fin n → α) (i : fin (n + 1)) : α :=
```

Comment:
This doesn't belong in the `list` namespace; the fact that it's implemented using lists is invisble to the user

Reply:
Good point. I've moved it to the `fin` namespace.